src/app/api/staff/mark-story/route.ts

```diff
@@ -0,0 +1,54 @@
+import prisma from "@/lib/prisma";
+import { getServerSession } from "next-auth";
+import { NextResponse, type NextRequest } from "next/server";
+import { patchSchema, postSchema } from "./schema";
+
+export async function POST(request: NextRequest) {
+  try {
+    const session = await getServerSession();
+    const user = await prisma.user.findUnique({
+      where: { email: session?.user.email ?? "noemail" },
+    });
+
+    if (!user || !user.roles.includes("EDITOR")) {
+      return NextResponse.json({ error: "Forbidden" }, { status: 403 });
+    }
+
+    const { description, story_id } = postSchema.parse(await request.json());
+    const story = await prisma.story.findFirst({ where: { id: story_id } });
+    const staffPickCheck = await prisma.staffPick.findUnique({
+      where: { storyId: story_id },
+    });
+    if (staffPickCheck) {
+      return NextResponse.json(
+        { error: "Story has already been Staff-picked" },
+        { status: 404 },
```

Comment:
is this best as status 404 or as 400 because the item technically doesn't not exist?

Reply:
I think 400 suit better. I will change it later